### PR TITLE
fix #1821

### LINKED ISSUE #1821: crash in spec plugin
## Steps to Reproduce the Problem

Create a file /usr/share/elektra/specification/default.ecf with the content:
```
[tests/tutorial/cascading/#0/current]
#@META override/#0 = /tests/overrides/test
test =
```

and a ~/.config/default.ecf with the content:
```
[]
```
(or more) and run any kdb command, e.g. `kdb ls /`.

## Expected Result

Usual behavior of command.

## Actual Result

Segmentation fault:
```
Program received signal SIGSEGV, Segmentation fault.
strlen () at ../sysdeps/x86_64/strlen.S:106
106     ../sysdeps/x86_64/strlen.S: Datei oder Verzeichnis nicht gefunden.
(gdb) bt
#0  strlen () at ../sysdeps/x86_64/strlen.S:106
#1  0x00007ffff6ca1c0f in __fnmatch (pattern=<optimized out>, pattern@entry=0x555555836100 "tests/tutorial/cascading/#0/current", string=<optimized out>, flags=flags@entry=1) at fnmatch.c:451
#2  0x00007ffff4e73cd8 in matchPatternToKey (key=0x555555834080, pattern=0x555555836100 "tests/tutorial/cascading/#0/current") at ./src/plugins/spec/spec.c:85
#3  doGlobbing (parentKey=parentKey@entry=0x55555580ec20, returned=returned@entry=0x5555558360c0, specKS=specKS@entry=0x555555835540, ch=ch@entry=0x55555583f2d0, dir=dir@entry=GET, clean=clean@entry=0) at ./src/plugins/spec/spec.c:680
#4  0x00007ffff4e74a00 in elektraSpecGet (handle=0x5555558407a0, returned=0x555555827ce0, parentKey=0x55555580ec20) at ./src/plugins/spec/spec.c:873
#5  0x00007ffff50793b3 in runPlugins (pluginKS=pluginKS@entry=0x55555583de40, modules=0x55555582a030, plugins=0x555555829f60, configOrig=0x55555583f5d0, returned=returned@entry=0x555555827ce0, parentKey=parentKey@entry=0x55555580ec20, op=GET, 
    traversalFunction=0x7ffff751f690 <ksNext>) at ./src/plugins/list/list.c:301
#6  0x00007ffff50796ca in elektraListGet (handle=<optimized out>, returned=0x555555827ce0, parentKey=0x55555580ec20) at ./src/plugins/list/list.c:337
#7  0x00007ffff772d871 in kdbGet (handle=<optimized out>, ks=0x555555827ce0, parentKey=0x55555580ec20) at ./src/libs/elektra/kdb.c:889
#8  0x0000555555583f80 in kdb::KDB::get (parentKey=..., returned=..., this=0x7fffffffd950) at ./src/bindings/cpp/include/kdb.hpp:183
#9  kdb::KDB::get (keyname="/sw/kdb/current/", returned=..., this=0x7fffffffd950) at ./src/bindings/cpp/include/kdb.hpp:166
#10 Cmdline::Cmdline (this=0x7fffffffe1b0, argc=3, argv=0x7fffffffe508, command=0x55555580e990) at ./src/tools/kdb/cmdline.cpp:302
#11 0x0000555555575196 in main (argc=3, argv=0x7fffffffe508) at ./src/tools/kdb/main.cpp:168
```

## System Information

- Elektra Version: master

### PR DESCRIPTION
**Pull request by tom-wa**
# Purpose

fix invalid handling of keyname causing a segfault

### CODE REVIEW

**Comment by markus2330** [comment]
jenkins build all please

**Comment by markus2330** [comment]
jenkins build mingw-w64 please

**Comment by markus2330** [comment]
jenkins build gcc-i386 please

**Comment by markus2330** [comment]
jenkins build mingw-w64 please

**Comment by markus2330** [comment]
jenkins build mingw please

**Comment by markus2330** [comment]
jenkins build multiconfig-gcc47-cmake-options please

File: src/plugins/spec/spec.c [hunk 1]
```diff
@@ -82,7 +82,11 @@ static char * keyNameToMatchingString (const Key * key)
 
 static int matchPatternToKey (const char * pattern, const Key * key)
 {
-	return !fnmatch (pattern, (strchr (keyName (key), '/') + 1), FNM_PATHNAME);
+	const char * ptr = strchr (keyName (key), '/');
```
Comment (markus2330) [inline src/plugins/spec/spec.c]: Can you please document a bit what matchPatternToKey does?

**Comment by markus2330** [comment]
@tom-wa Please do not forget to also write something in the release notes in doc/news/_preparation_next_release.md for any change.

**Comment by ingwinlu** [comment]
jenkins build all please

**Comment by markus2330** [comment]
@waht Do you know what happened here?

```
 35/116 MemCheck  #54: testio_uv ........................***Failed    6.89 sec
src/libs/io/io.c:263:elektraIoBindingAddIdle: operation cannot be assigned to multiple bindings
src/libs/io/io.c:212:elektraIoBindingAddTimer: operation cannot be assigned to multiple bindings
src/libs/io/io.c:161:elektraIoBindingAddFd: operation cannot be assigned to multiple bindings
```

**Comment by waht** [comment]
@markus2330 this should be normal logging output: the test verifies that an operation cannot be assigned to multiple bindings. The reason the test failed is the [timeout later on](https://build.libelektra.org/jenkins/job/elektra-gcc-configure-debian-log/1518/console):
`test_fd.c:87: error in testFdShouldSignalXControl: timeout; test failed`

**Comment by markus2330** [comment]
Thank you, maybe you can improve the output a bit (like "checking if operation cannot be assigned...").

Do you have an idea why that timeout occurred?

**Comment by ingwinlu** [comment]
> Do you have an idea why that timeout occurred?

the node might have been under heavy load as I was running a 'stress' test for the jenkins master. I queued a rerun to see if it occurs again when there is less load.

**Comment by ingwinlu** [comment]
there we go, test passed.

**Comment by markus2330** [comment]
Why is it down to 7 checks now?

**Comment by ingwinlu** [comment]
hard to pinpoint. probably something overwrote the status when I triggered the single build.

**Comment by markus2330** [comment]
jenkins build all please

**Comment by ingwinlu** [comment]
jenkins build all please

**Comment by ingwinlu** [comment]
jenkins build all please

**Comment by markus2330** [comment]
jenkins build all please

**Comment by markus2330** [comment]
Thank you for the fix!

**Pull request merged**